src/main/java/com/google/gcloud/storage/Blob.java

```diff
@@ -157,12 +157,12 @@ public Builder acl(List<Acl> acl) {
       return this;
     }
 
-    public Builder owner(Acl.Entity owner) {
+    Builder owner(Acl.Entity owner) {
```

Comment:
I think it would be clearer to drop the setters for the immutable fields - just set them in the constructor and make them final.

This whole class doesn't seem to add much over the underlying proto. It also seems to be fulfilling a confused role, sometimes being just the id (bucket, path), and sometimes being a descriptor with all the metadata.